internal/services/devcenter/dev_center_catalogs_resource.go

```diff
@@ -0,0 +1,305 @@
+package devcenter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devcenter/2023-04-01/catalogs"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type DevCenterCatalogsResourceModel struct {
+	Name              string `tfschema:"name"`
+	ResourceGroupName string `tfschema:"resource_group_name"`
+	DevCenterID       string `tfschema:"dev_center_id"`
+	CatalogType       string `tfschema:"catalog_type"`
+	URI               string `tfschema:"uri"`
+	Branch            string `tfschema:"branch"`
+	KeyVaultKeyUrl    string `tfschema:"key_vault_key_url"`
+	Path              string `tfschema:"path"`
+}
+
+type DevCenterCatalogsResource struct{}
+
+var _ sdk.Resource = DevCenterCatalogsResource{}
+
+func (r DevCenterCatalogsResource) ModelObject() interface{} {
+	return &DevCenterCatalogsResourceModel{}
+}
+
+func (r DevCenterCatalogsResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return catalogs.ValidateCatalogID
+}
+
+func (r DevCenterCatalogsResource) ResourceType() string {
+	return "azurerm_dev_center_catalog"
+}
+
+func (r DevCenterCatalogsResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Required:     true,
+			Type:         pluginsdk.TypeString,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"dev_center_id": {
+			Required:     true,
+			Type:         pluginsdk.TypeString,
+			ForceNew:     true,
+			ValidateFunc: catalogs.ValidateDevCenterID,
+		},
+
+		"catalog_type": {
+			Required: true,
+			Type:     pluginsdk.TypeString,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"gitHub",
+				"adoGit",
```

Comment:
Based on the design of this resource these want to be two blocks which conflict with each other, one for `github` and one for `azure_devops_git` - or two entirely separate resources - rather than trying to top-level these fields, can we update this to use two blocks instead?

Reply:
Split to 2 separate blocks as requested.